src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -83,12 +83,23 @@ abstract class Kernel implements KernelInterface, RebootableInterface, Terminabl
     const END_OF_MAINTENANCE = '07/2019';
     const END_OF_LIFE = '01/2020';
 
-    public function __construct(string $environment, bool $debug)
+    public function __construct(string $environment, bool $debug, string $projectDir = null)
     {
         $this->environment = $environment;
         $this->debug = $debug;
         $this->rootDir = $this->getRootDir(false);
         $this->name = $this->getName(false);
+        if (null !== $projectDir && !is_dir($projectDir)) {
```

Comment:
I think we should remove this check: it's overhead on 100% of requests

Reply:
I agree, that's done now. It's less descriptive but should not be a problem anyway because it's optional and people may only use it if they know what they are doing